packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -1025,6 +1025,76 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
     return signatureWithModuleAddress;
   }
 
+  /**
+   * Deploys the smart contract
+   *
+   * This method will deploy a Smart Account contract. It is useful for deploying in a moment when you know that gas prices are low,
+   * and you want to deploy the account before sending the first user operation. This step can otherwise be skipped,
+   * as the deployment will alternatively be bundled with the first user operation.
+   *
+   * @param buildUseropDto {@link BuildUserOpOptions}.
+   * @returns Promise<{@link UserOpResponse}> that you can use to track the user operation.
+   * @error Throws an error if the account has already been deployed.
+   * @error Throws an error if the account has not enough native token balance to deploy, if not using a paymaster.
+   *
+   * @example
+   * import { createClient } from "viem"
+   * import { createSmartAccountClient } from "@biconomy/account"
+   * import { createWalletClient, http } from "viem";
+   * import { polygonMumbai } from "viem/chains";
+   *
+   * const signer = createWalletClient({
+   *   account,
+   *   chain: polygonMumbai,
+   *   transport: http(),
+   * });
+   *
+   * const smartAccount = await createSmartAccountClient({
+   *  signer,
+   *  biconomyPaymasterApiKey,
+   *  bundlerUrl
+   * });
+   *
+   * // If you want to use a paymaster...
+   * const { wait } = await smartAccount.deploy({
+   *   paymasterServiceData: { mode: PaymasterMode.SPONSORED },
+   * });
+   *
+   * // Or if you can't use a paymaster send native token to this address:
+   * const counterfactualAddress = await smartAccount.getAccountAddress();
+   *
+   * // Then deploy the account
+   * const { wait } = await smartAccount.deploy();
+   *
+   * const { success, receipt } = await wait();
+   *
+   */
+  public async deploy(buildUseropDto?: BuildUserOpOptions): Promise<UserOpResponse> {
+    const accountAddress = this.accountAddress ?? (await this.getAccountAddress());
+
+    // Check that the account has not already been deployed
+    const byteCode = await this.provider?.getBytecode({ address: accountAddress as Hex });
+    if (byteCode !== undefined) {
+      throw new Error(ERROR_MESSAGES.ACCOUNT_ALREADY_DEPLOYED);
+    }
+
+    // Check that the account has enough native token balance to deploy, if not using a paymaster
+    if (!buildUseropDto?.paymasterServiceData?.mode) {
+      const nativeTokenBalance = await this.provider?.getBalance({ address: accountAddress });
+      if (nativeTokenBalance === BigInt(0)) {
+        throw new Error(ERROR_MESSAGES.NO_NATIVE_TOKEN_BALANCE_DURING_DEPLOY);
+      }
+    }
+
+    return this.sendTransaction(
+      {
+        to: accountAddress,
```

Comment:
this will still encode the userop.calldata right?

for this we would want userOp.callData to go '0x'

Reply:
Ah I see, makes sense, will change. I would have expected tests to fail though 🤔

Reply:
no it would still work but unnecessarily would consume more gas and we intend to Just deploy.